cmake/TestSetup.cmake

```diff
@@ -11,6 +11,43 @@
 # the public, perform publicly and display publicly, and to permit others to do so.
 #=========================================================================================
 
+
+
+# Search for the python interpreter
+message(STATUS "Searching for python 3.6 or higher, required to run regression tests.")
+find_package(Python3 3.6 QUIET COMPONENTS Interpreter)
+
+# search executable path
+if ( NOT Python3_FOUND)
```

Comment:
Which system required the following code to manually implement the find for Python?

Reply:
Give me a few minutes.

Reply:
ping @nmsriram you added this section.

Reply:
We discussed on the phone - sounds like Cmake 3.11 and earlier did not have `FindPython`. However, it did have `FindPythonInterp`, so we should use that instead for older versions.

Reply:
@AndrewGaspar @JoshuaSBrown CI is now failing with:
````
61CMake Error at CMakeLists.txt:14 (cmake_minimum_required):
62  CMake 3.12 or higher is required.  You are running version 3.10.3
````

Reply:
Yeah, I'm proposing something like:
```cmake
if (CMAKE_VERSION VERSION_LESS "3.12")
    find_package(Python 3.6 REQUIRED)
else()
    find_package(PythonInterp 3.6 REQUIRED)
    set(Python_Interpreter_FOUND ON)
    set(Python_EXECUTABLE ${PYTHON_EXECUTABLE})
endif()
```

Reply:
I'm working on this will update within the next 20 minutes.

Reply:
Ok, that took far longer than I thought. This works as expected.